pkg/controller/namespace/namespace_controller.go

```diff
@@ -112,11 +112,16 @@ func (nm *NamespaceController) worker() {
 			if err := nm.syncNamespaceFromKey(key.(string)); err != nil {
 				if estimate, ok := err.(*contentRemainingError); ok {
 					go func() {
+						defer utilruntime.HandleCrash()
 						t := estimate.Estimate/2 + 1
 						glog.V(4).Infof("Content remaining in namespace %s, waiting %d seconds", key, t)
 						time.Sleep(time.Duration(t) * time.Second)
 						nm.queue.Add(key)
 					}()
+				} else {
+					// rather than wait for a full resync, re-add the namespace to the queue to be processed
+					nm.queue.Add(key)
```

Comment:
Actually wait (I mean not wait with this PR, it's correct, but wait with  #22045).

Now I think, it that was something different, because even if we don't readd it here, we will sec the namespace every resync period, which is now 5 minutes:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-controller-manager/app/options/options.go#L61

If that's the case, if we didn't remove the namespace for 1h30m (which was the case in that issue), this would mean that we had errors that we didn't handled correctly ~100 times, which seems quite non-realistic.
@derekwaynecarr - am I missing something?